elegant/gui/general_pose_annotator.py

```diff
@@ -0,0 +1,89 @@
+import pathlib
+import pickle
+from PyQt5 import Qt
+import numpy
+
+from ris_widget import ris_widget
+from elegant.gui import pose_annotation
+from elegant import worm_widths
+from elegant import worm_spline
+
+class GeneralPoseAnnotator:
+    def __init__(self, rw, pixels_per_micron=1/1.3):
+        self.ris_widget = rw
+        width_estimator = worm_widths.WidthEstimator.from_default_widths(pixels_per_micron=pixels_per_micron)
+        self.pose_annotator = pose_annotation.PoseAnnotation(self.ris_widget, width_estimator=width_estimator)
+        self.ris_widget.add_annotator([self.pose_annotator])
+        
+        load_data = Qt.QPushButton('Load')
+        load_data.clicked.connect(self.load_annotations)
+        save_data = Qt.QPushButton('Save')
+        save_data.clicked.connect(self.save_annotations)
+        self.pose_annotator._add_row(self.pose_annotator.widget.layout(), Qt.QLabel('Data:'),load_data, save_data)
+
+    def save_annotations(self):
+        """Save the pose annotations as pickle files into the parent directory.
+        A pickle file is created for each page in the flipbook with the name of the first image in the 
+        flipbook_page list as the base for the pickle file name.
+        """
+        for fp in self.ris_widget.flipbook_pages:
+            annotations = getattr(fp, 'annotations', None)
+            if annotations is not None:
+                pose = annotations.get('pose')
+                if pose is not None and pose is not (None, None):
+                    path = pathlib.Path(fp[0].name)
+                    lab_frame_image = fp[0].data
+                    center_tck, width_tck = pose
+                    #warp and save the warped images
+                    if center_tck is not None:
+                        warp = worm_spline.to_worm_frame(lab_frame_image, center_tck, width_tck)
+                        warp_save_path = path.parent/(path.stem+"-straight.png")
+                        freeimage.write(warp, warp_save_path)
+
+                        #if the widths are drawn, then create a mask that allows the user to make an alpha channel later
+                        if width_tck is not None:
+                            worm_frame_mask = worm_spline.worm_frame_mask(width_tck, warp.shape)
+                            img_save_path = path.parent/(path.stem+"-mask.png")
+                            freeimage.write(worm_frame_mask, img_save_path)
+
+                    #save annotations into a pickle file
+                    save_path = path.with_suffix('.pickle')
+                    with save_path.open('wb') as f:
+                        pickle.dump(annotations, f)
+
+    def load_annotations(self):
+        """Loads annotation metadata from a pickle file for each flipbook page.
+        NOTE: If there is already a pose annotation, the metadata will not be loaded.
+        """
+        for fp in self.ris_widget.flipbook_pages:
+            annotations = getattr(fp, 'annotations', None)
```

Comment:
instead of `None` if the default for `getattr` is an empty dict (i.e. `{}`) then the next line will work fine. Otherwise you'll first have to make sure that `annotations` is in fact a dict, and not `None`.